AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/editor_script_canvas.py

```diff
@@ -0,0 +1,119 @@
+
+        returns a string for the component property path
+        """
+
+        component_property_path = self.PROPERTIES
+
+        match variable_state:
+            case VariableState.UNUSEDVARIABLE:
+                component_property_path += self.UNUSED_VARIABLES
+            case VariableState.VARIABLE:
+                component_property_path += self.VARIABLES
+
+        component_property_path += variable_name + "|"
+        component_property_path += self.DATUM_PATH
+        component_property_path += variable_name
+
+        return component_property_path
+
+class ScriptCanvasComponent:
+
+    def __init__(self, editor_entity: EditorEntity):
+        self.script_canvas_component = editor_entity.add_component(SCRIPT_CANVAS_UI)
+
+    def get_script_canvas_component(self) -> EditorComponent:
+        return self.script_canvas_component
+
+    def set_component_graph_file_from_path(self, sc_file_path: str) -> None:
+        """
+        Function for updating the script canvas file being used by the component
+
+        param sc_file_path : the string path the new graph file on disk
+
+        returns None
+        """
+        sourcehandle = scriptcanvas.SourceHandleFromPath(sc_file_path)
+
+        self.script_canvas_component.set_component_property_value(Path.SCRIPT_CANVAS_ASSET_PROPERTY, sourcehandle)
+
+    def get_component_graph_file(self) -> str:
+        """
+        function for getting the component property value of the graph file in the script canvas component
+
+        returns the component property value of the script canvas source handle (string)
+        """
+
+        return self.script_canvas_component.get_component_property_value(Path.SCRIPT_CANVAS_COMPONENT_PROPERTY_PATH)
+
+    def set_variable_value(self, variable_name: str, variable_state: VariableState, variable_value) -> str:
+        """
+        Function for setting a value to a variable on the script canvas component.
+
+        Note: The variable must be exposed to the component. this can be done through the script canvas node inspector.
+        variables exposed to the component will be visible by name rather than type.
+
+        param variable_name: the name of the variable
+        param variable_state: used/unused variable
+        param variable_value: the value to assign to the variable
+
+        returns component property path as a string
+        """
+        component_property_path = Path.construct_variable_component_property_path(Path, variable_name, variable_state)
+
+        self.script_canvas_component.set_component_property_value(component_property_path, variable_value)
+
+        return component_property_path
+
+    def get_variable_value(self, variable_name: str, variable_state: VariableState):
+        """
+        Function for getting the value of an exposed variable off the script canvas component. The variable type
+        can vary but only 2 types are currently supported: String adn Boolean. See github GH-13344 for more info.
```

Comment:
Update GHI reference to the o3de/o3de#13344 format and make it the start of a line discussing the issue so it's up front and center to somebody reading this docstring that there's an ongoing implementation issue.

Reply:
updated this and the setter as well since clients using either should be aware of the current limitations